stl/inc/inplace_vector

```diff
@@ -0,0 +1,1560 @@
+        _STL_VERIFY(_Pos < _My_size, "inplace_vector subscript out of range");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+        return _My_data[_Pos];
+    }
+
+    _NODISCARD constexpr _Ty& at(const size_type _Pos) {
+        if (_Pos >= _My_size) {
+            _Xrange();
+        }
+
+        return _My_data[_Pos];
+    }
+
+    _NODISCARD constexpr const _Ty& at(const size_type _Pos) const {
+        if (_Pos >= _My_size) {
+            _Xrange();
+        }
+
+        return _My_data[_Pos];
+    }
+
+    _NODISCARD constexpr _Ty& front() noexcept /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_My_size != 0, "front() called on empty inplace_vector");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+        return _My_data[0];
+    }
+
+    _NODISCARD constexpr const _Ty& front() const noexcept /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_My_size != 0, "front() called on empty inplace_vector");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+        return _My_data[0];
+    }
+
+    _NODISCARD constexpr _Ty& back() noexcept /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_My_size != 0, "back() called on empty inplace_vector");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+        return _My_data[_My_size - 1];
+    }
+
+    _NODISCARD constexpr const _Ty& back() const noexcept /* strengthened */ {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_My_size != 0, "back() called on empty inplace_vector");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+
+        return _My_data[_My_size - 1];
+    }
+
+    // [containers.sequences.inplace.vector.data], data access
+    _NODISCARD constexpr _Ty* data() noexcept {
+        return _My_data;
+    }
+
+    _NODISCARD constexpr const _Ty* data() const noexcept {
+        return _My_data;
+    }
+
+    // [containers.sequences.inplace.vector.modifiers], modifiers
+    template <class... _Valty>
+    constexpr _Ty& emplace_back(_Valty&&... _Val) {
+        return _Emplace_back(_STD forward<_Valty>(_Val)...);
+    }
+
+    constexpr _Ty& push_back(const _Ty& _Val) {
+        return _Emplace_back(_Val);
+    }
+
+    constexpr _Ty& push_back(_Ty&& _Val) {
+        return _Emplace_back(_STD move(_Val));
+    }
+
+    template <_Container_compatible_range<_Ty> _Rng>
+    constexpr void append_range(_Rng&& _Range) {
+        if constexpr (/* _RANGES forward_range<_Rng> || */ _RANGES sized_range<_Rng>) {
+            const auto _Length = _STD _To_unsigned_like(_RANGES distance(_Range));
+            const auto _Count  = _STD _Convert_size<size_type>(_Length);
+            _Append_counted_range(_RANGES _Ubegin(_Range), _Count);
+        } else {
+            _Append_uncounted_range(_RANGES _Ubegin(_Range), _RANGES _Uend(_Range));
+        }
+    }
+
+    constexpr void pop_back() noexcept /* strengthened */ {
+#if _ITERATOR_DEBUG_LEVEL == 2
+        _STL_VERIFY(_My_size != 0, "inplace_vector empty before pop");
+#endif // _ITERATOR_DEBUG_LEVEL == 2

```

Comment:
See #4586. I think  we should use `_CONTAINER_DEBUG_LEVEL`.

And I think you can improve `<vector>` in another PR.